app/models/space.rb

```diff
@@ -116,6 +116,18 @@ def self.roles
     Space::USER_ROLES.map { |r| Role.find_by_name(r) }
   end
 
+  def last_activitity
+    RecentActivity.where(owner: self).last
+  end
+
+  # Order by when the last activity in the space happened
+  scope :order_by_activity, -> {
```

Comment:
Actually, we should also consider the activities in the webconf rooms. This is one possible solution:

```ruby
# note: we need this "*", otherwise it will select only spaces.* and the result will be different
Space.select("*")
  .uniq
  .joins(:bigbluebutton_room)
  .joins("LEFT JOIN activities ON ((activities.owner_type = 'Space' AND activities.owner_id = spaces.id) OR (activities.owner_type = 'BigbluebuttonRoom' AND activities.owner_id = bigbluebutton_rooms.id))")
  .order("activities.created_at DESC")
```

A bit more verbose but faster. It was the best way I found to use a single SQL and still return a relation that can be paginated.
The number of activities in the db should grow pretty fast, so having a single SQL like this one filtering everything seems good.